### PR TITLE
[AIRFLOW-1743] Verify ldap filters correctly

### PR DESCRIPTION
**Pull request by bolkedebruin**
Dear Airflow maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [X] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "[AIRFLOW-XXX] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-1743

### Description
- [X] Here are some details about my PR, including screenshots of any UI changes:

The superuser and data profiler filter where set by default
in the config template and could not be unset.

### Tests
- [X] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits
- [X] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

cc @criccomini

### CODE REVIEW

**Comment by criccomini** [comment]
cc @jgao54 @wrp

File: airflow/contrib/auth/backends/ldap_auth.py [hunk 1]
```diff
@@ -175,7 +194,7 @@ def try_login(username, password):
 
         search_scope = LEVEL
         if configuration.has_option("ldap", "search_scope"):
-            search_scope = SUBTREE if configuration.get("ldap", "search_scope") == "SUBTREE" else LEVEL
+            search_scope = SUBTREE if (configuration.get("ldap", "search_scope") == "SUBTREE" else LEVEL
```
Comment (criccomini) [inline airflow/contrib/auth/backends/ldap_auth.py]: Is this extra paren supposed to be here?

Comment (bolkedebruin) [reply airflow/contrib/auth/backends/ldap_auth.py]: no ill remove that

**Comment by criccomini** [comment]
LGTM

**Pull request closed without merging**